rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/monitoring/views/compact/Tile.java

```diff
@@ -0,0 +1,357 @@
+                    CompactView.getGlobalHover().hide();
+                }
+                dirty = false;
+                if (node != null) {
+                    setHoverNodeLabel();
+                } else if (host != null) {
+                    hoverLabel.setContents("<strong>Host</strong><br>" + host.getHostName());
+                } else if (nodesource != null) {
+                    hoverLabel.setContents("<strong>NodeSource</strong><br>" + nodesource.getSourceName());
+                }
+                hover.moveTo(event.getClientX() - 155, event.getClientY() - 65);
+                hover.show();
+                CompactView.setGlobalHover(hover);
+            }
+        });
+    }
+
+    private void addClickHandler() {
+        this.addClickHandler(event -> {
+            compactView.setDoNotScroll(true);
+            if (node != null) {
+                compactView.getController().selectNode(node);
+            } else if (host != null) {
+                compactView.getController().selectHost(host);
+            } else if (nodesource != null) {
+                compactView.getController().selectNodeSource(nodesource);
+            }
+        });
+    }
+
+    private void addMouseOutHandler() {
+        this.addMouseOutHandler(event -> {
+            dirty = true;
+            hover.hide();
+            CompactView.setGlobalHover(null);
+        });
+    }
+
+    public void refresh(Node n) {
+        this.node = n;
+        this.setUrl(n.getIcon());
+        this.setHoverNodeLabel();
+    }
+
+    public void refresh(NodeSource ns) {
+        this.nodesource = ns;
+        this.setUrl(ns.getIcon());
+    }
+
+    private void setHoverNodeLabel() {
+        String str = "Provider: " + node.getNodeProvider() + "<br>";
+        if (node.getNodeOwner().trim().length() > 0) {
+            str += "Used by: " + node.getNodeOwner() + "<br>";
+        }
+        str += node.getNodeState().toString() + " since " + JSUtil.getTime(node.getTimeStamp());
+        this.hoverLabel.setContents(str);
+    }
+
+    void setSelectedTile(boolean selected) {
+        if (selected) {
+            this.getElement().getStyle().setBackgroundColor("#7caaf7");
+            this.getElement().getStyle().setBorderColor("#d9e4f6");
+            this.highlightGroup(true);
+        } else {
+            this.getElement().getStyle().setBackgroundColor("white");
+            this.getElement().getStyle().setBorderColor("white");
+            this.highlightGroup(false);
+        }
+    }
+
+    private void highlightGroup(boolean selected) {
+        // highlight every node following this one in the compactPanel layout,
+        // until we hit one from another host/ns
+        if (host != null || nodesource != null) {
+            int id;
+            if (host == null) {
+                id = panel.indexOf(nodesource).get();
+            } else {
+                id = panel.indexOf(host).get();
+            }
+
+            while (true) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS135)